middlewares/osio/auth.go

```diff
@@ -39,8 +42,8 @@ type SecretLocator interface {
 }
 
 type cacheData struct {
-	Token    string
-	Location string
+	Token     string
+	Namespace namespace
```

Comment:
Instead of Location (having only cluster.APIURL) now full object of Namespace is cached which has all the URLs like cluster-api-url, metrics-url, console-url.